spk/firebird/src/service-setup.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+FBGUARD_BIN_FILE="${SYNOPKG_PKGDEST}/bin/fbguard"
+
+# We set onetime because we want the systemd to take care about the failures
+SERVICE_COMMAND="${FBGUARD_BIN_FILE} -pidfile ${PID_FILE} -daemon -onetime"
+
+SYSDBA_PASSWORD_FILE="${SYNOPKG_PKGDEST}/SYSDBA.password"
+
```

Comment:
This file must be moved to the var folder. Otherwise it will not survive a package update.
```suggestion
SYSDBA_PASSWORD_FILE="${SYNOPKG_PKGVAR}/SYSDBA.password"

```

Reply:
Is it correct to assume that all the `.conf` files belong to this location too?

Reply:
@publicarray, @th0ma7 please help with the `etc` folder.
I am used to place all configuration data into the `var` folder, since we have built in save and restore for all DSM versions.

IMHO the `etc` folder will be lost on package updates on DSM 6 but not on DSM 7 (?) except the specific file installer-variables that is used to keep wizard variables that are kept via wizard dialogs for updates in all DSM versions.

Reply:
Moved the SYSDBA.password to ${SYNOPKG_PKGVAR}

Reply:
There was an intent a little more than two years ago to create installation prefixes for all other synology default directorie https://github.com/SynoCommunity/spksrc/pull/4579

We ended-up settling on the `var` only.  Doing the same thing for `etc` would be a no-brainer but would have an impact on many packages as this would mean defaulting to:
```
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
```
And we would need to confirm exact behavior of that directory on DSM6 and DSM7 pertaining to upgrades...  Along with potentially re-adjusting `service-setup.sh` scripts who may now have to point to `etc` for its configuration files.

Long story short, probably safe to keep things in `var` for now?

Reply:
Ok, I will move the config to var. Is there some common way how to handle conflicts between user modified configuration and new configuration provided by updated package? E.g. apt allows you to choose during upgrade whether you want the updated package version or your modified version.
The DSM packages upgrades are AFAIK not interactive, so we should always preserve the users version?

Reply:
> Ok, I will move the config to var. Is there some common way how to handle conflicts between user modified configuration and new configuration provided by updated package? E.g. apt allows you to choose during upgrade whether you want the updated package version or your modified version. The DSM packages upgrades are AFAIK not interactive, so we should always preserve the users version?

Our generic service installer handles this, so that customized config files that are created in the installation process are not overwridden by a package update.

On DSM 7 the var folder in the package and the var folder that is used after installation are not at the same location. The installer has to move/copy those to the real var folder. When updating an existing package under DSM 7 the files in the var folder are renamed to *.new to avoid overwriting existing files but provide the updated version of the files with another name. 
This helps the users to update existing configuration files that might have changed format.

On DSM 6 the var folder is backuped and restored to avoid overwriting existing files in the var folder. If a package wants to provide the newer configuration files renamed to *.new, then it must provide respective code in service-setup.sh. This is not handled by the generic package installer.

Reply:
This is neat. What is the proper way of populating the `$(STAGING_INSTALL_PREFIX_VAR)`? Should I do that manually in the install_target? Shouldn't the PLIST related magic be supposed to do this? The PLIST however cares only about the files in `/var/packages/firebird/target` and whatever is under `/var/packages/firebird/var/` must be done manually?

Reply:
Aha, that is neat. I will move the configuration files. At the moment changing the conf dir path breaks the installation and I have to figure out why this happens.

Reply:
Ok it done now. It required some hint in the firebird mailing lists. The latest version however installs its configuration to PKGVAR and preserves the conf files across upgrades.